tests/pipelines/lint/test_nfcore_yml.py

```diff
@@ -43,15 +43,87 @@ def test_nfcore_yml_fail_repo_type(self):
     def test_nfcore_yml_fail_nfcore_version(self):
         """Lint test: nfcore_yml - FAIL - nf-core version not set"""
 
-        with open(self.nf_core_yml) as fh:
-            content = fh.read()
-            new_content = re.sub(r"nf_core_version:.+", "nf_core_version: foo", content)
-        with open(self.nf_core_yml, "w") as fh:
-            fh.write(new_content)
-        lint_obj = nf_core.pipelines.lint.PipelineLint(self.new_pipeline)
-        lint_obj._load()
-        results = lint_obj.nfcore_yml()
+        self.nf_core_yml["nf_core_version"] = "foo"
+        self.yaml.dump(self.nf_core_yml, self.nf_core_yml_path)
+        assert self.lint_obj._load()
+        results = self.lint_obj.nfcore_yml()
         assert "nf-core version in `.nf-core.yml` is not set to the latest version." in str(results["warned"])
         assert len(results.get("failed", [])) == 0
         assert len(results.get("passed", [])) >= 0
         assert len(results.get("ignored", [])) == 0
+
+    def test_nfcore_yml_nested_lint_config(self) -> None:
+        """Lint test: nfcore_yml with nested lint config - PASS"""
+        valid_yaml = """
+        lint:
+            files_unchanged:
+                - .github/workflows/branch.yml
+            # modules_config: False
+            modules_config:
+                    - fastqc
+            # merge_markers: False
+            merge_markers:
+                    - docs/my_pdf.pdf
+            # nextflow_config: False
+            nextflow_config:
+                - manifest.name
+                - config_defaults:
+                    - params.annotation_db
+                    - params.multiqc_comment_headers
+                    - params.custom_table_headers
+            multiqc_config:
+                - report_section_order
+                - report_comment
+            files_exist:
+                - .github/CONTRIBUTING.md
+                - CITATIONS.md
+            # template_strings: False
+            template_strings:
+                    - docs/my_pdf.pdf
+        """
+        self.nf_core_yml["lint"] = self.yaml.load(valid_yaml)
+        self.yaml.dump(self.nf_core_yml, self.nf_core_yml_path)
+
+        assert self.lint_obj._load()
+        results = self.lint_obj.nfcore_yml()
+        assert len(results.get("failed", [])) == 0
+        assert len(results.get("warned", [])) == 0
+        assert len(results.get("ignored", [])) == 0
```

Comment:
Why can these be all equal to 0? I would assume that we would have some ignored?

Reply:
we only lint the `.nf-core.yml` file here, so the above settings don't ignore anything there.